partner_time_to_pay/tests/test_partner_time_to_pay.py

```diff
@@ -0,0 +1,72 @@
+#   (<http://www.serpentcs.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import datetime
+from odoo.tests.common import TransactionCase
+from dateutil.relativedelta import relativedelta
+
+
+class TestPartnerTimeToPay(TransactionCase):
+
+    def setUp(self):
+        super(TestPartnerTimeToPay, self).setUp()
+        self.payment_model = self.env['account.payment']
+        self.register_payments_model = self.env['account.register.payments']
+        self.partner_id = self.env.ref('base.res_partner_1')
+        self.product_id = self.env.ref('product.product_product_4')
+        self.account_model = self.env['account.account']
+        self.account_invoice_model = self.env['account.invoice']
+        self.account_journal = self.env['account.journal']
+        self.payment_method_manual_in =\
+            self.env.ref("account.account_payment_method_manual_in")
+        self.account_receivable_id =\
+            self.env.ref('account.data_account_type_receivable')
+        self.account_revenue_id =\
+            self.env.ref('account.data_account_type_revenue')
+
+        self.account_receivable = self.account_model.search(
+            [('user_type_id', '=', self.account_receivable_id.id)], limit=1)
+        self.account_revenue = self.account_model.search(
+            [('user_type_id', '=', self.account_revenue_id.id)], limit=1)
+
+        self.invoice_id = self.account_invoice_model.create({
+            'partner_id': self.partner_id.id,
+            'type': 'out_invoice',
+            'account_id': self.account_receivable.id,
+            'date_invoice': datetime.datetime.now().date(),
+            'invoice_line_ids': [(0, 0, {
+                'product_id': self.product_id.id,
+                'name': self.product_id.name,
+                'price_unit': 100,
+                'quantity': 10,
+                'account_id': self.account_revenue.id,
+            })]
+        })
+
+        self.invoice_id._onchange_invoice_line_ids()
+        self.invoice_id._onchange_cash_rounding()
+        self.invoice_id.action_invoice_open()
+        self.bank_journal_euro =\
+            self.account_journal.create({
+                'name': 'Bank',
+                'type': 'bank',
+                'code': 'BNK67'
+            })
+
+        ctx = {
+            'active_model': 'account.invoice',
+            'active_ids': [self.invoice_id.id]
+        }
+        self.register_payments =\
+            self.register_payments_model.with_context(ctx).create({
+                'payment_date':
+                    datetime.datetime.now().date() + relativedelta(days=10),
+                'journal_id': self.bank_journal_euro.id,
+                'payment_method_id': self.payment_method_manual_in.id,
+            })
+        self.register_payments.create_payments()
+        self.payment = self.payment_model.search([], order="id desc", limit=1)
+
+    def test_partner_compute_d2x(self):
+        self.partner_id._compute_d2x()
```

Comment:
@PieterPaulussen Test case was run already, And Runbot is Successfully started.
So no need to double check.

Can you please update your review?

Reply:
It is not 'double' checking. You don't have any validation on the results of you compute function.

Reply:
@PieterPaulussen Ok, I will add validation for computing

Reply:
Sorry, perhaps you misunderstood my intentions with validation.
I was thinking something more along the lines of:
```python
self.assertEqual(self.partner_id.d2p_life, 10.0)
self.assertEqual(self.partner_id.d2p_ytd, 10.0)
self.assertEqual(self.partner_id.d2r_life, 0.0)
self.assertEqual(self.partner_id.d2r_ytd, 0.0)
```
At the moment you are still just checking that the calculation occurs, not that the results of the calculations are correct.

Reply:
@PieterPaulussen Can you please suggest the best logic or Create new PR for that, It's Not a big Deal :smile:

Reply:
@PieterPaulussen You can suggest how to achieve our Goal (Computing validation).
Because here many misunderstanding with US.
Can you please give me the right direction.

Reply:
I'm not sure how more specific I can be. Replace your test method with this:
```python
    def test_partner_compute_d2x(self):
        self.assertEqual(self.partner_id.d2p_life, 10.0)
        self.assertEqual(self.partner_id.d2p_ytd, 10.0)
        self.assertEqual(self.partner_id.d2r_life, 0.0)
        self.assertEqual(self.partner_id.d2r_ytd, 0.0)
```